encrypted-media/scripts/waiting-for-a-key.js

```diff
@@ -0,0 +1,175 @@
+function runTest(config)
+{
+    // For debugging timeouts, keep track of the number of the
+    // various events received.
+    var debugEncryptedEventFired = false;
+    var debugWaitingForKeyEventFired = false;
+    var debugTimeUpdateEventCount = 0;
+    var debugMessage = '';
+
+    // Set global option explicit_timeout to true and control
+    // the timeout in the promise test below.
+    setup({
+        explicit_timeout: true
+    });
+
+    promise_test(function (test) {
+        var video = config.video;
+        var keysystem = config.keysystem;
+        var configuration = {
+            initDataTypes: [config.initDataType],
+            audioCapabilities: [{
+                contentType: config.audioType
+            }],
+            videoCapabilities: [{
+                contentType: config.videoType
+            }],
+            sessionTypes: ['temporary']
+        };
+        var initData;
+        var initDataType;
+        var mediaKeySession;
+        // As this code doesn't wait for the 'message' event for clearkey to avoid
+        // race conditions with 'waitingforkey', specify the key ID and
+        // key used by the encrypted content.
+        var keyId = new Uint8Array(config.content.keys[0].kid);
+        var rawKey = new Uint8Array(config.content.keys[0].key);
+        // Use the message handler for non clearkey drm
+        var handler = config.messageHandler || null;
+
+        test.timeout = function () {
+            var message = 'timeout. message = ' + debugMessage
+              + ', encrypted: ' + debugEncryptedEventFired
+              + ', waitingforkey: ' + debugWaitingForKeyEventFired
+              + ', timeupdate count: ' + debugTimeUpdateEventCount;
+            // It seems there is a bug in testharness: According to the function name and testharness
+            // docs: " it is possible to use test.force_timeout() in place of assert_unreached(),
+            // to immediately fail the test but with a status of TIMEOUT. ", but it doesn't fail the test
+            // immediately. If this will be fixed, then nothing after force_timeout will be executed.
+            // Need to check this again once the bug is fixed
+            test.force_timeout();
```

Comment:
Why do you need force_timeout? Could you just set phase manually, since you are calling set_status anyway(it is already called in force_timeout)? Also seems like there is a bug in testharness. According to the function name and testharness docs: " it is possible to use test.force_timeout() in place of assert_unreached(), to immediately fail the test but with a status of TIMEOUT. ", but it doesn't fail the test immediately. If this will be fixed, then nothing after force_timeout will be executed.